app/controllers/translated_attributes_controller.rb

```diff
@@ -16,25 +18,48 @@ def destroy
   private
 
   def create_translated_attr
-    a = TranslatedAttribute.create!(permitted_params)
-    head :no_content, location: "translated_attributes/#{a.id}"
+    a = @resource.translated_attributes.create(permitted_params)
+
+    render_translation_attribute_error(a) and return if a.errors.any?
+
+    head :no_content, location: "/resources/#{@resource.id}/translated-attributes/#{a.id}"
   end
 
   def update_translated_attr
-    load_translated_attr.update!(permitted_params)
+    a = load_translated_attr
+    a.update(permitted_params)
+
+    render_translation_attribute_error(a) and return if a.errors.any?
+
     head :no_content
   end
 
+  def render_translation_attribute_error(a)
+    if a.errors[:key] == ["has already been taken"]
+      render(json: {errors: {"code" => "key_already_exists"}}, status: 400)
+    elsif a.errors[:key] == ["can't be blank"]
+      render(json: {errors: {"code" => "invalid_key"}}, status: 400)
+    elsif a.errors[:onesky_phrase_id] == ["can't be blank"]
+      render(json: {errors: {"code" => "invalid_onesky_phrase_id"}}, status: 400)
```

Comment:
There currently isn't consistency in the `onesky-phrase-id` attribute name. when reading the value from json the name is: `onesky-phrase-id`, when creating a new `translated-attribute` or updating an existing one the attribute name needs to be `onesky_phrase_id`.

These attribute names should be consistent between reads and writes. I would prefer the attribute name be `onesky-phrase-id` for API interactions with `translated-attribute` objects to follow the `kebab-naming` convention.

I do know we have similar inconsistencies in other APIs that exist, at some point we should probably clean those up, but that's out of the scope of this ticket and PR.

Reply:
when submitting this create request:

```
POST /resources/3/translated-attributes HTTP/1.1
Host: mobile-content-api-stage.cru.org
Content-Type: application/vnd.api+json
Accept: application/vnd.api+json
Content-Length: 175

{
    "data": {
        "type": "translated-attribute",
        "attributes": {
            "key": "outline2",
            "onesky-phrase-id": "info.outline"
        }
    }
}
```

I get back this error:
```
{
    "errors": {
        "code": "invalid_onesky_phrase_id"
    }
}
```